crates/view/src/planner.rs

```diff
@@ -843,3 +843,178 @@ impl<R: RngCore + CryptoRng> Planner<R> {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use penumbra_asset::STAKING_TOKEN_ASSET_ID;
+    use penumbra_keys::keys::{Bip44Path, SeedPhrase, SpendKey};
+    use rand_core::OsRng;
+
+    #[test]
+    fn test_sufficient_funds_for_non_zero_fees_in_transaction() {
+        let rng = OsRng;
+        let seed_phrase = SeedPhrase::generate(rng);
+        let sk = SpendKey::from_seed_phrase_bip44(seed_phrase, &Bip44Path::new(0));
+        let fvk = sk.full_viewing_key();
+        let (sender, _dtk) = fvk.incoming().payment_address(0u32.into());
+
+        let seed_phrase_2 = SeedPhrase::generate(rng);
+        let sk_2 = SpendKey::from_seed_phrase_bip44(seed_phrase_2, &Bip44Path::new(0));
+        let fvk_2 = sk_2.full_viewing_key();
+        let (reciever, _dtk_2) = fvk_2.incoming().payment_address(0u32.into());
+
+        let note0 = Note::generate(
+            &mut OsRng,
+            &sender,
+            Value {
+                amount: 3u64.into(),
+                asset_id: *STAKING_TOKEN_ASSET_ID,
+            },
+        );
+        let note1 = Note::generate(
+            &mut OsRng,
+            &sender,
+            Value {
+                amount: 5u64.into(),
+                asset_id: *STAKING_TOKEN_ASSET_ID,
+            },
+        );
+
+        let mut spendable_notes: Vec<Note> = Vec::new();
+        spendable_notes.push(note0);
+        spendable_notes.push(note1);
+
+        let mut planner = Planner::new(OsRng);
+
+        // Set non-zero gas price.
+        let mut gas_price = GasPrices::default();
+        gas_price.block_space_price = 1u64;
+        let fee_tier = FeeTier::Low;
+
+        planner.set_gas_prices(gas_price).set_fee_tier(fee_tier);
+
+        let amount = Value {
+            amount: 3u64.into(),
+            asset_id: *STAKING_TOKEN_ASSET_ID,
+        };
+
+        planner.output(amount, reciever);
+        planner.actions = planner.plan.actions.clone();
+
+        let mut iterations = 0usize;
+        while let Some(_required) = planner
```

Comment:
I think adding tests is a great idea, but my sense is that rather than replicating the logic (which has changed since this PR was opened) we should either:
- refactor the core planning logic into a pure function
- check in on #3913 to see if the plan is to be able to build a mock view client at some point, in which case we can use `Planner::plan` directly

Reply:
My intuition is that it's probably best to wait a little longer until we try to test this, we'll definitely have less turbulence in the planner in a couple weeks time or less